virttest/env_process.py

```diff
@@ -1083,6 +1084,13 @@ def preprocess(test, params, env):
                         int(params.get("pre_command_timeout", "600")),
                         params.get("pre_command_noncritical") == "yes")
 
+    # Sysprep the master image if requested, to customize image before cloning
+    if params.get("sysprep_required", "no") == "yes":
+        logging.info("Syspreping the image as requested before cloning.")
+        image_filename = storage.get_image_filename(params, base_dir)
+        sysprep_options = params.get("sysprep_options", "--operations machine-id")
+        utils_libguestfs.virt_sysprep_cmd(image_filename, options=sysprep_options)
```

Comment:
it would be good to handle the failure case `utils_libguestfs.virt_sysprep_cmd`, may be we can keep inside `try..catch` block and error out/proceed based on some flags?